internal/include/otel_types.h

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _OTEL_TYPES_H
+#define _OTEL_TYPES_H
+
+#include "go_types.h"
+#include "common.h"
+
+/* Defintions should mimic structs defined in go.opentelemetry.io/otel/attribute */
+typedef u64 attr_val_type_t;
+
+#define INVALID 0
+#define BOOL 1
+#define INT64 2
+#define FLOAT64 3
+#define STRING 4
+#define BOOLSLICE 5
+#define INT64SLICE 6
+#define FLOAT64SLICE 7
+#define STRINGSLICE 8
+
+typedef struct go_otel_attr_value {
+	attr_val_type_t  vtype;
+	u64              numeric;
+	struct go_string string;
+	struct go_iface	 slice;
+} go_otel_attr_value_t;
+
+typedef struct go_otel_key_value {
+	struct go_string     key;
+	go_otel_attr_value_t value;
+} go_otel_key_value_t;
+
+#define OTEL_ATTRIBUTE_KEY_MAX_LEN      (32)
+#define OTEL_ATTRIBUTE_VALUE_MAX_LEN    (128)
+#define OTEL_ATTRUBUTE_MAX_COUNT        (16)
+
+typedef struct otel_attirbute {
+	u16 val_length;
+	u8 vtype;
+	u8 reserved;
+	char key[OTEL_ATTRIBUTE_KEY_MAX_LEN];
+	char value[OTEL_ATTRIBUTE_VALUE_MAX_LEN];
+} otel_attirbute_t;
+
+typedef struct otel_attributes {
+	otel_attirbute_t attrs[OTEL_ATTRUBUTE_MAX_COUNT];
+	u8 valid_attrs;
+}__attribute__((packed)) otel_attributes_t;
+
+static __always_inline bool set_attr_value(otel_attirbute_t *attr, go_otel_attr_value_t *go_attr_value)
+{
+	switch (go_attr_value->vtype)
+	{
+	case BOOL:
+	case INT64:
+	case FLOAT64:
+		bpf_probe_read(&attr->value, sizeof(s64), &go_attr_value->numeric);
+		return true;
+	case STRING:
+		if (go_attr_value->string.len <= 0){
+			return false;
+		}
+		if (go_attr_value->string.len >= OTEL_ATTRIBUTE_VALUE_MAX_LEN) {
+			bpf_printk("Aattribute string value is too long\n");
+			return false;
+		}
+		if (!get_go_string_from_user_ptr(&go_attr_value->string, attr->value, OTEL_ATTRIBUTE_VALUE_MAX_LEN)) {
```

Comment:
Could this be simplified to the following?

```suggestion
		return get_go_string_from_user_ptr(&go_attr_value->string, attr->value, OTEL_ATTRIBUTE_VALUE_MAX_LEN)
```